tests/cli/mod.rs

```diff
@@ -0,0 +1,44 @@
+use assert_cmd::prelude::*; // Add methods on commands
+use seafowl::config::schema::DEFAULT_SQLITE_DB;
+use std::fs::File;
+use std::io::{BufRead, BufReader, Write};
+use std::path::Path;
+use std::process::{Command, Stdio}; // Run programs
+use tempfile::{Builder, TempDir};
+
+mod basic;
+
+const TEST_CONFIG_FILE: &str = "seafowl-test.toml";
+
+fn setup_temp_config_and_data_dir() -> std::io::Result<TempDir> {
```

Comment:
> Hm, we don't have any of this as an existing fixture in integration tests? I guess we always inject a pre-created context instead of setting up a config file for Seafowl.

That is correct, we didn't have this before. Typically we use 
https://github.com/splitgraph/seafowl/blob/3d428c98bd8b83237a3cb6cae1bb48080a3c4026/tests/statements/mod.rs#L50-L53
to get a Seafowl context object directly.

> Is there a specific reason we have to use on-disk state here instead of in-memory? Doesn't seem like the CLI tests care about persistence when restarting the CLI anyway.

Yes there is; in particular the test starts up the Seafowl bin as an external command, so there's no way to supply anything programmatically to it directly, short of saving a config file on disk and then supplying the config file via the `-c` arg
```rust
let temp_dir = setup_temp_config_and_data_dir()?;
    let mut cmd = Command::cargo_bin("seafowl").expect("Seafowl bin exists");
    cmd.arg("--cli")
        .arg("-c")
        .arg(temp_dir.path().join(TEST_CONFIG_FILE))
        .stdin(Stdio::piped())
        .stdout(Stdio::piped())
        .stderr(Stdio::piped());
```

I guess an alternative is to extend `make_context_with_pg` such that it always generates a temp dir (not only in the case of `ObjectStoreType::Local` as now), and then also persist the generated config file there.

Reply:
> I guess an alternative is to extend make_context_with_pg such that it always generates a temp dir (not only in the case of ObjectStoreType::Local as now), and then also persist the generated config file there.

Ok, I'll go ahead with the merge, and then potentially address this in the follow-up PRs.